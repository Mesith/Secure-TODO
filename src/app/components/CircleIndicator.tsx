import React from "react";
import { View, StyleSheet } from "react-native";

const CircleIndicator = () => {
  return <View style={styles.circleIndicator} />;
};

const styles = StyleSheet.create({
  circleIndicator: {
    height: 24,
    width: 24,
    borderRadius: 24 / 2,
    marginHorizontal: 8,
    backgroundColor: "#053095",
  },
});

export default CircleIndicator;
