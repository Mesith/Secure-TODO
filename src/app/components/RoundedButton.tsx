import React from "react";
import { TouchableOpacity, Text, StyleSheet } from "react-native";

type RounderedButtonProps = {
  text: string;
  width: number;
  onPress: () => void;
};
// eslint-disable-next-line react/display-name
const RoundedButton = React.memo(
  ({ text, width, onPress }: RounderedButtonProps) => {
    return (
      <TouchableOpacity
        testID="rounded-button"
        style={{ width, ...styles.button }}
        onPress={onPress}
      >
        <Text style={styles.text}>{text}</Text>
      </TouchableOpacity>
    );
  }
);

const styles = StyleSheet.create({
  button: {
    height: 40,
    backgroundColor: "#053095",
    alignItems: "center",
    justifyContent: "center",
    borderRadius: 16,
  },
  text: {
    textAlign: "center",
    color: "#FFFFFF",
  },
});

export default RoundedButton;
