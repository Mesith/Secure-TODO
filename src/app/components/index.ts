export { default as AddOrUpdateTodoView } from "./AddOrUpdateTodoView";
export { default as CircleIndicator } from "./CircleIndicator";
export { default as RoundedButton } from "./RoundedButton";
export { default as TodoItem } from "./TodoItem";
