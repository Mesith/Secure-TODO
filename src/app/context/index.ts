export { AuthContext } from "./AuthContext";
export { DBContext } from "./DBContext";
export { AuthContextProvider } from "./AuthContext";
export { DBContextProvider } from "./DBContext";
