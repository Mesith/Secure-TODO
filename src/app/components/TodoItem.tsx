import React from "react";
import {
  View,
  StyleSheet,
  Text,
  TouchableOpacity,
  TouchableWithoutFeedback,
} from "react-native";
import CircleIndicator from "./CircleIndicator";

type TodoListItemProps = {
  name: string;
  onTodoSelect: () => void;
  onTodoDelete: () => void;
};

const TodoItem = ({ name, onTodoSelect, onTodoDelete }: TodoListItemProps) => {
  return (
    <TouchableOpacity style={styles.container} onPress={onTodoSelect}>
      <View style={styles.nameContainer}>
        <CircleIndicator />
        <Text>{name}</Text>
      </View>
      <TouchableWithoutFeedback testID="remove-todo-button">
        <TouchableOpacity style={styles.removeButton} onPress={onTodoDelete}>
          <Text style={styles.removeButtonText}>REMOVE</Text>
        </TouchableOpacity>
      </TouchableWithoutFeedback>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    height: 64,
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    backgroundColor: "#FFFFFF",
    marginBottom: 10,
    borderRadius: 20,
  },
  nameContainer: {
    height: 64,
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
  },
  text: {
    textAlign: "center",
    color: "#FFFFFF",
  },
  removeButton: {
    marginRight: 10,
  },
  removeButtonText: {
    color: "#757474",
  },
});

export default TodoItem;
