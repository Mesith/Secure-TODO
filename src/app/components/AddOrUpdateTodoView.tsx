import React, { useCallback, useEffect, useState } from "react";
import { StyleSheet, TextInput, View } from "react-native";
import useDBContext from "../hooks/UseDBContext";
import RoundedButton from "./RoundedButton";

const AddOrUpdateTodoView = () => {
  const { currentTodoItem, addTodo, updateTodo } = useDBContext();
  const [name, setName] = useState("");
  const [id, setId] = useState<number | null>(null);

  useEffect(() => {
    if (currentTodoItem) {
      const { id, name } = currentTodoItem;
      setName(name);
      setId(id);
    } else {
      setName("");
      setId(null);
    }
  }, [currentTodoItem]);

  const handleSaveOrUpdate = useCallback(() => {
    const trimmedName = name.trim();
    if (!trimmedName) {
      return;
    }

    if (id) {
      updateTodo({ id, name: trimmedName });
    } else {
      addTodo(trimmedName);
    }

    setName("");
    setId(null);
  }, [name, id, addTodo, updateTodo]);

  return (
    <View style={styles.container}>
      <TextInput
        style={styles.input}
        value={name}
        onChangeText={setName}
        placeholder="Enter here"
      />
      <RoundedButton
        width={80}
        text={id ? "Update" : "Add"}
        onPress={handleSaveOrUpdate}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    backgroundColor: "#FFFFFF",
    alignItems: "center",
    width: "94%",
    height: 64,
    padding: 8,
    borderRadius: 16,
  },
  input: {
    flex: 1,
    height: 44,
    paddingHorizontal: 8,
    textAlign: "left",
    marginRight: 4,
    borderBottomWidth: 1,
    borderBottomColor: "#efefef",
  },
});

export default AddOrUpdateTodoView;
