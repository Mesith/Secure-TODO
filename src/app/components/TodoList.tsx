import React from "react";
import { StyleSheet, FlatList } from "react-native";
import { Todo } from "../context/DBContext";
import TodoItem from "./TodoItem";

type TodoListProps = {
  todos: Todo[];
  handleTodoSelect: (item: Todo) => void;
  handleTodoDelete: (item: Todo) => void;
};

// eslint-disable-next-line react/display-name
const TodoList = React.memo(
  ({ todos, handleTodoSelect, handleTodoDelete }: TodoListProps) => {
    return (
      <FlatList
        data={todos}
        renderItem={({ item }: { item: Todo }) => (
          <TodoItem
            name={item?.name}
            onTodoSelect={() => handleTodoSelect(item)}
            onTodoDelete={() => handleTodoDelete(item)}
          />
        )}
        keyExtractor={(item: Todo) => item.id.toString()}
        style={styles.listStyle}
      />
    );
  }
);

const styles = StyleSheet.create({
  listStyle: {
    width: "95%",
    marginBottom: 16,
    marginTop: 16,
  },
});

export default TodoList;
