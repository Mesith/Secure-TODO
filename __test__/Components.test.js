import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import { RoundedButton, TodoItem } from "../src/app/components";

describe("Testing components", () => {
  test("Test RoundedButton, matches snapshot", () => {
    const { toJSON } = render(
      <RoundedButton text="Submit" width={100} onPress={() => {}} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  test("Renders RoundedButton with correct text and width", () => {
    const { getByText, getByTestId } = render(
      <RoundedButton text="Submit" width={100} onPress={() => {}} />
    );
    const button = getByTestId("rounded-button");
    const buttonText = getByText("Submit");

    expect(button).toBeTruthy();
    expect(buttonText).toBeTruthy();
    expect(button.props.style).toHaveProperty("width", 100);
  });

  test("Testing TodoItem , renders the name text and handles remove button click", () => {
    const name = "Test Todo";
    const onTodoSelectMock = jest.fn();
    const onTodoDeleteMock = jest.fn();

    const { getByText, getByTestId } = render(
      <TodoItem
        name={name}
        onTodoSelect={onTodoSelectMock}
        onTodoDelete={onTodoDeleteMock}
      />
    );

    const nameText = getByText(name);
    expect(nameText).toBeDefined();

    const removeButton = getByTestId("remove-todo-button");
    fireEvent.press(removeButton);
    expect(onTodoDeleteMock).toHaveBeenCalledTimes(1);
  });
});
