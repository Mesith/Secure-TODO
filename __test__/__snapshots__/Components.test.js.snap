// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing components Test RoundedButton, matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#053095",
      "borderRadius": 16,
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "width": 100,
    }
  }
  testID="rounded-button"
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "textAlign": "center",
      }
    }
  >
    Submit
  </Text>
</View>
`;
